lib/fastly_nsq/ssl_context.rb

```diff
@@ -0,0 +1,44 @@
+class SSLContext
+  def initialize(context = nil)
+    @context = context || {}
```

Comment:
I might be wrong but I believe you can use a NullObject here, like:

```ruby
@context = context || NullSSLContext.new
```

And have that just respond to `.to_h` with a `nil`.